src/Illuminate/Console/Scheduling/ScheduleGroup.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Illuminate\Console\Scheduling;
+
+use Closure;
+use DateTimeZone;
+
+class ScheduleGroup
+{
+    use ManagesFrequencies;
+
+    /**
+     * The cron expression representing the grouped events' frequency.
+     */
+    protected string $expression = '* * * * *';
+
+    /**
+     * How often to repeat the grouped events during a minute.
+     */
+    protected ?int $repeatSeconds = null;
+
+    /**
+     * The timezone the grouped events' date should be evaluated on.
+     */
+    protected DateTimeZone|string $timezone;
+
+    /**
+     * The user the grouped events should run as.
+     */
+    protected ?string $user;
+
+    /**
+     * The list of environments the grouped events should run under.
+     *
+     * @var string[]
+     */
+    protected array $environments;
+
+    /**
+     * Indicates if the grouped events should run in maintenance mode.
+     */
+    protected bool $evenInMaintenanceMode;
+
+    /**
+     * Indicates if the grouped events should not overlap itself.
+     */
+    protected bool $withoutOverlapping;
+
+    /**
+     * Indicates if the grouped events should only be allowed to run on one server for each cron expression.
+     */
+    protected bool $onOneServer;
+
+    /**
+     * The number of minutes the grouped events' mutex should be valid.
+     */
+    protected int $expiresAt;
+
+    /**
+     * Indicates if the grouped events should run in the background.
+     */
+    protected bool $runInBackground;
+
+    /**
+     * The array of filter callbacks.
+     *
+     * @var array<int, Closure|bool>
+     */
+    protected array $filters = [];
+
+    /**
+     * The array of reject callbacks.
+     *
+     * @var array<int, Closure|bool>
+     */
+    protected array $rejects = [];
+
+    public function __construct(
+        protected Schedule $schedule,
+        protected Closure $onRegister
+    ) {
+    }
+
+    public function schedules(callable $callback): void
+    {
+        $callback($this->schedule);
+
+        ($this->onRegister)();
+    }
+
+    public function mergeAttributes(Event $event): void
+    {
+        $event->expression = $this->expression;
+        $event->repeatSeconds = $this->repeatSeconds;
+
+        if (isset($this->withoutOverlapping) && $this->withoutOverlapping) {
+            $event->withoutOverlapping($this->expiresAt);
+        }
+
+        // Merge the filter callbacks into the event.
+        foreach ($this->filters as $filter) {
+            $event->when($filter);
+        }
+
+        // Merge the reject callbacks into the event.
+        foreach ($this->rejects as $reject) {
+            $event->skip($reject);
+        }
+
+        /**
+         * Loop through the attributes and only set the ones that were set on the group.
+         */
```

Comment:
Inconsistent comment format:

```suggestion
        // Loop through the attributes and only set the ones that were set on the group.
```